### PR TITLE
New climber

### PR DESCRIPTION
**Pull request by OfriRosenbaum**

### CODE REVIEW

**Review by tal-sitton — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/climbing/MoveToNextBar.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+public class MoveToNextBar extends SequentialCommandGroup {
+
+    public MoveToNextBar() {
+        ClimberWinch winch = ClimberWinch.getInstance();
+        ClimberPlacer leftPlacer = ClimberPlacer.getLeftInstance();
+        ClimberPlacer rightPlacer = ClimberPlacer.getRightInstance();
+        addCommands(
+                new MoveGenericSubsystem(winch, ClimberWinch.UP_SPEED).withInterrupt(() -> winch.getEncoderPosition() <=
+                        winch.ENCODER_RELEASE_BAR_POSITION.get()),
+                new ParallelCommandGroup(
```
[the remark this answers is not in the document]
Comment (OfriRosenbaum) [reply src/main/java/frc/robot/commands/climbing/MoveToNextBar.java:28]: This is already in it (?)

File: src/main/java/frc/robot/subsystems/ClimberPlacer.java [hunk 2]
```diff
@@ -15,14 +17,28 @@
  */
 public class ClimberPlacer extends MotoredGenericSubsystem {
 
-    public static final double MIN_SPEED = -0.6;
+    public static final double MIN_SPEED = -0.1;
     public static final double MAX_SPEED = 0.6;
 
-    private final Supplier<Double> ENCODER_VELOCITY_TOLERANCE =
-            rootNamespace.addConstantDouble("velocity tolerance", 0);
+    public final Supplier<Double> ENCODER_DOWN_POSITION =
```
[the remark this answers is not in the document]
Comment (OfriRosenbaum) [reply src/main/java/frc/robot/subsystems/ClimberPlacer.java:26]: image

File: src/main/java/frc/robot/commands/climbing/MovePlacerToNextBar.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package frc.robot.commands.climbing;
+
+import com.spikes2212.command.genericsubsystem.commands.MoveGenericSubsystem;
+import frc.robot.subsystems.ClimberPlacer;
+
+/**
+ * Moves a {@code ClimberPlacer} until it reaches the next bar.
```
Comment (tal-sitton) [inline src/main/java/frc/robot/commands/climbing/MovePlacerToNextBar.java]: `@link`

File: src/main/java/frc/robot/commands/climbing/MovePlacerToNextBar.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * @see ClimberPlacer
```
Comment (tal-sitton) [inline src/main/java/frc/robot/commands/climbing/MovePlacerToNextBar.java]: remove

File: src/main/java/frc/robot/commands/climbing/MoveToNextBar.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import frc.robot.subsystems.ClimberWinch;
+
+/**
+ * Moves the robot to the next bar. <br>
+ * Steps: <br>
+ * 1. Moves the winch up to release the hooks from the bar.<br>
+ * 2. Moves the placers downwards.<br>
+ * 3. Extends the winch to it's full extent.<br>
+ * 4. Raises the placers until they hit the next bar and go into a stall.<br>
+ * 5. Closes the winch while keeping the placers next to the bar to move the robot to the next bar.<br>
```
Comment (tal-sitton) [inline src/main/java/frc/robot/commands/climbing/MoveToNextBar.java]: change

**Review by tal-sitton — CHANGES_REQUESTED** [review]

**Review by tal-sitton — APPROVED** [review]

**Pull request merged**